manifests/mount.pp

```diff
@@ -83,8 +83,8 @@
     fail("Unknown option ${ensure} for ensure")
   }
 
-  $mount_options = [ $options, $ll, $lf, $t, $dim, $r, ]
-  $_options = join(delete($mount_options, ''), ',')
+  $mount_options = [ $options, $ll, $lf, $dim, $r, ]
+  $_options = join(delete_undef_values(delete($mount_options, '')), ',')
```

Comment:
This function was introduced in stdlib 4.2.0, as mentioned in my PR to address the same problem: #39.  The functionality of undefined variables changes between Puppet 3.x and 4.x, so `delete_undef_values()` only works for Puppet 4.x.  I don't think it makes sense at this time to require Puppet 4.x for this module, which is why my PR simply makes explicitly empty strings for each of the mount options.